### PR TITLE
chore: refactor gen subcommands

### PR DESCRIPTION
**Pull request by AlekSi**
Move them to own package.

### CODE REVIEW

File: cmd/talosctl/cmd/mgmt/gen/csr.go [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+		opts := []x509.Option{}
+
+		parsed := net.ParseIP(csrIP)
+		if parsed == nil {
+			return fmt.Errorf("invalid IP: %s", csrIP)
+		}
+
+		ips := []net.IP{parsed}
+		opts = append(opts, x509.IPAddresses(ips))
+		opts = append(opts, x509.NotAfter(time.Now().Add(time.Duration(crtHours)*time.Hour))) // BUG
```
Comment (AlekSi) [inline cmd/talosctl/cmd/mgmt/gen/csr.go:59]: `crtHours` is always zero for `csr` command. Should this line be dropped, or there should be flag for that property?

Comment (smira) [reply cmd/talosctl/cmd/mgmt/gen/csr.go:59]: I think these commands are largely not really supported atm

Comment (AlekSi) [reply cmd/talosctl/cmd/mgmt/gen/csr.go:59]: Well, they are documented: https://www.talos.dev/docs/v0.11/guides/managing-pki/

I will drop that line then.

Comment (smira) [reply cmd/talosctl/cmd/mgmt/gen/csr.go:59]: I know they are documented, but they're not really sitting in the regular usage pattern

**Comment by AlekSi** [comment]
Will re-create

**Pull request closed without merging**